src/testRunner/unittests/tsserverProjectSystem.ts

```diff
@@ -8830,4 +8847,108 @@ export const x = 10;`
             assert.equal(moduleInfo.cacheSourceFile.sourceFile.text, updatedModuleContent);
         });
     });
+
+    function makeSampleProjects() {
+        const aTs: File = {
+            path: "/a/a.ts",
+            content: "export function fnA() {}",
+        };
+        const aTsconfig: File = {
+            path: "/a/tsconfig.json",
+            content: `{
+                "compilerOptions": {
+                    "outDir": "bin",
+                    "declaration": true,
+                    "declarationMap": true,
+                    "composite": true,
+                }
+            }`,
+        };
+
+        const bTs: File = {
+            path: "/b/b.ts",
+            content: 'import { fnA } from "../a/a";\nexport function fnB() { fnA(); }',
+        };
+        const bTsconfig: File = {
+            path: "/b/tsconfig.json",
+            content: `{
+                "compilerOptions": {
+                    "outDir": "bin",
+                },
+                "references": [
+                    { "path": "../a" }
+                ]
+            }`,
+        };
+
+        const host = createServerHost([aTs, aTsconfig, bTs, bTsconfig]);
+        const session = createSession(host);
+
+        writeDeclarationFiles(aTs, host, session, [
+            { name: "/a/bin/a.d.ts.map", text: '{"version":3,"file":"a.d.ts","sourceRoot":"","sources":["../a.ts"],"names":[],"mappings":"AAAA,wBAAgB,GAAG,SAAK"}' },
+            // Need to mangle the sourceMappingURL part or it breaks the build
+            { name: "/a/bin/a.d.ts", text: `export declare function fnA(): void;\n//# source${""}MappingURL=a.d.ts.map` },
+        ]);
+
+        return { session, aTs, bTs };
+    }
+
+    describe("tsserverProjectSystem project references", () => {
+        it("goToDefinition", () => {
+            const { session, aTs, bTs } = makeSampleProjects();
+
+            openFilesForSession([bTs], session);
+
+            const definitionRequest = makeSessionRequest<protocol.FileLocationRequestArgs>(CommandNames.Definition, protocolFileLocationFromSubstring(bTs, "fnA()"));
+            const definitionResponse = session.executeCommand(definitionRequest).response as protocol.DefinitionResponse["body"];
+
+            assert.deepEqual(definitionResponse, [protocolFileSpanFromSubstring(aTs, "fnA")]);
+        });
+
+        it("navigateTo", () => {
+            const { session, bTs } = makeSampleProjects();
+
+            openFilesForSession([bTs], session);
+
+            const navtoRequest = makeSessionRequest<protocol.NavtoRequestArgs>(CommandNames.Navto, { file: bTs.path, searchValue: "fn" });
+            const navtoResponse = session.executeCommand(navtoRequest).response as protocol.NavtoResponse["body"];
+
+            assert.deepEqual(navtoResponse, [
+                // TODO: First result should be from a.ts, not a.d.ts

```

Comment:
This is just a reminder because `navTo` doesn't apply sourcemaps yet, right?

Reply:
Right, working on it.